homework_9/main.py

```diff
@@ -0,0 +1,66 @@
+"""
+Напишите следующие функции:
+* Нахождение корней квадратного уравнения
+* Генерация csv файла с тремя случайными числами в каждой строке. 100-1000 строк.
+* Декоратор, запускающий функцию нахождения корней квадратного уравнения с каждой тройкой чисел из csv файла.
+* Декоратор, сохраняющий переданные параметры и результаты работы функции в json файл.
+"""
+import csv
+import json
+from pathlib import Path
+from random import randint as rnd
+
+CSV_PATH = Path.cwd() / 'numbers.csv'
+JSON_PATH = Path.cwd() / 'result.json'
+
+
+def deco_quadratic(func):
+    def wrapper():
+        csv_generator()
+        with open(CSV_PATH) as f1:
+            data = csv.reader(f1, quotechar=',', quoting=csv.QUOTE_NONNUMERIC)
+            for i, row in enumerate(data, 1):
+                res = func(*row)
+                res = res if res else "No solutions"
+                save_to_json({i: {'Parameters': row, 'Results': res}})
+        return res
+
+    return wrapper
+
+
+def save_to_json(upd: dict):
+    if JSON_PATH.is_file():
+        with open(JSON_PATH) as f2:
+            data = json.load(f2)
+    else:
+        data = {}
+
+    data.update(upd)
+
+    with open(JSON_PATH, 'w') as f3:
+        json.dump(data, f3, indent=2)

```

Comment:
А вот так можно переделать эту функцию в декоратор.
```suggestion
def save_to_json(func):
    if JSON_PATH.is_file():
        with open(JSON_PATH, encoding='utf-8') as f2:
            data = json.load(f2)
    else:
        data = []
    def wrapper(*args)
        res = func(*args)
        data.append({'Parameters': args, 'Results': res})
        with open(JSON_PATH, 'w', encoding='utf-8') as f3:
            json.dump(data, f3, indent=2)
```